tests/test_callbacks.py

```diff
@@ -1,101 +1,93 @@
-                    )
-
-                    path = os.path.join(
-                        os.path.dirname(__file__),
-                        'screenshots',
-                        dir_name,
-                        name + '.png'
-                    )
-
-                    self.driver.save_screenshot(path)
-
-        def click_button_and_save(name_to_xpaths, dir_name, save=True):
-            for name, xpath in name_to_xpaths.items():
-                button = self.driver.find_element(By.XPATH, xpath)
-                button.click()
-
-                if save:
-                    WebDriverWait(self.driver, 20).until(
-                        EC.presence_of_element_located((By.ID, "cytoscape"))
-                    )
-
-                    path = os.path.join(
-                        os.path.dirname(__file__),
-                        'screenshots',
-                        dir_name,
-                        name + '.png'
-                    )
-
-                    self.driver.save_screenshot(path)
-
-        create_input_and_save(
+        # Create directory if it doesn't already exist
+        if not os.path.exists(directory_path):
+            os.makedirs(directory_path)
+
+        if prefix and not name_map:
+            name_map = {
+                option: prefix + option
+                for option in options
+            }
+
+        elif not name_map:
+            name_map = {}
+
+        for option in options:
+            elem.send_keys(Keys.CONTROL + "a")
+            elem.send_keys(option)
+            elem.send_keys(Keys.RETURN)
+
+            if save:
+                # If the name map doesn't contain a custom name for the option,
+                # we default to the value of the option to name the saved
+                # screenshot
+                name = name_map.get(option, option)
+                name = name.replace('(', '_').replace(')', '').replace(',', '_')
+                name = name.replace('#', '_hex_').replace(' ', '')
+
+                self.dash_duo.wait_for_element_by_id("cytoscape", 20)
+
+                path = os.path.join(
+                    os.path.dirname(__file__),
+                    'screenshots',
+                    dir_name,
+                    name + '.png'
+                )
+
+                self.dash_duo.driver.save_screenshot(path)
+
+    def click_button_and_save(self, name_to_xpaths, dir_name, save=True):
+        for name, xpath in name_to_xpaths.items():
+            button = self.dash_duo.driver.find_element(By.XPATH, xpath)
+            button.click()
+
+            if save:
+                self.dash_duo.wait_for_element_by_id("cytoscape", 20)
+
+                path = os.path.join(
+                    os.path.dirname(__file__),
+                    'screenshots',
+                    dir_name,
+                    name + '.png'
+                )
+
+                self.dash_duo.driver.save_screenshot(path)
+
+    def test_callbacks(self, dash_duo):
```

Comment:
Our usual usage of `dash.testing` does not create classes at all - just functions, and they should include test case ids, for which our convention is 4 characters to identify the file and three numbers for the test case within the file. So this one might be `test_cycb001_callbacks(dash_duo)`, in `test_interactions.py` they might be `test_cyin001_dragging`, `test_cyin002_clicking` etc... the advantage here is you can easily run exactly the test you want using for example `pytest -k cyin002`.

Helper functions like `create_input_and_save` can either be nested inside the test case (if they're only used in one place like in this file) or they can be explicitly passed `dash_duo`